Jobs/Submissions/DeleteSubmissionFileJob.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file Jobs/Submissions/DeleteSubmissionFileJob.php
```

Comment:
I see a bit of inconsistency with job naming ("Delete" vs. "Deleted" in `DeletedSubmissionSearchJob` vs. `DeleteSubmissionFileJob`). Likewise below for events (`DeleteSubmissionFile` vs. `DeletedSubmission`).

Since the code that triggers the event will handle doing the named action (e.g. deleting a submission file) outside of the event and job, we should be careful not to suggest that it's the event and job that do the task. So maybe this makes sense:

- Event names: Name these for what has happened and use past tense (e.g. `SubmissionDeleted`)
- Job names: Name these for what the job itself is doing (e.g. `RemoveSubmissionFileIndex`)

I'm not so sure about a convention for the listener. For example, currently there's one called `DeleteSubmissionFileListener` and it associates the event and the job. If there are several jobs, the same listener will be used to associate all of them, correct?

Reply:
> I'm not so sure about a convention for the listener. For example, currently there's one called DeleteSubmissionFileListener and it associates the event and the job. If there are several jobs, the same listener will be used to associate all of them, correct?

Yes.

Reply:
OK, then it makes sense to me that the listener is named for the thing that happened (e.g. `SubmissionDeleted`), rather than what needs to happen as a result.